test/unit/math.spec.ts

```diff
@@ -98,6 +123,35 @@ export class MathTests {
         Expect(result).toBe(expected);
     }
 
+    @TestCase("++of().p.d", "o=4;a=6")
+    @TestCase("of().p.d++", "o=4;a=6")
+    @TestCase("--of().p.d", "o=2;a=6")
+    @TestCase("of().p.d--", "o=2;a=6")
+    @TestCase("of().p.d += af()[i()][i()]", "o=9;a=6")
+    @TestCase("of().p.d -= af()[i()][i()]", "o=-3;a=6")
+    @TestCase("of().p.d *= af()[i()][i()]", "o=18;a=6")
+    @TestCase("af()[i()][i()] /= of().p.d", "o=3;a=2.0")
```

Comment:
This seems to be exactly what I would expect, maybe it would be nice to also check the other elements in a have not changed though, maybe.

Reply:
Good point